server/src/infra/repositories/access.repository.ts

```diff
@@ -219,6 +219,14 @@ export class AccessRepository implements IAccessRepository {
         },
       });
     },
+    hasActivityAccess: (_: string, albumId: string): Promise<boolean> => {
+      return this.albumRepository.exist({
+        where: {
+          id: albumId,
+          isActivityEnabled: true,
+        },
+      });
+    },
```

Comment:
Can you just merge this into the exist access checks? It can become something like:

```typescript
album: {
  id: albumId,
  isActivityEnabled: true,
}
```

Reply:
for `album.hasOwnerAccess` and `album.hasSharedAlbumAccess` ?

Reply:
Oh right, I guess that is used in other places as well. You're right!